gateway-api-integrator/lib/charms/gateway_api_integrator/v0/gateway_route.py

```diff
@@ -0,0 +1,702 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""gateway-route interface library.
+
+## Getting Started
+
+To get started using the library, you just need to fetch the library using `charmcraft`.
+
+```shell
+cd some-charm
+charmcraft fetch-lib charms.gateway_api_integrator.v0.gateway_route
+```
+
+In the `metadata.yaml` of the charm, add the following:
+
+```yaml
+requires:
+    gateway-route:
+        interface: gateway-route
+        limit: 1
+```
+
+Then, to initialise the library:
+
+```python
+from charms.gateway_api_integrator.v0.haproxy_route import GatewayRouteRequirer
+
+class SomeCharm(CharmBase):
+  def __init__(self, *args):
+    # ...
+
+    # There are 2 ways you can use the requirer implementation:
+    # 1. To initialize the requirer with parameters:
+    self.gateway_route_requirer = GatewayRouteRequirer(self,
+        relation_name=<required>,
+        name=<optional>,
+        model=<optional>,
+        port=<optional>,
+        paths=<optional>,
+        hostname=<optional>,
+    )
+
+    # 2.To initialize the requirer with no parameters, i.e
+    # self.gateway_route_requirer = GatewayRouteRequirer(self)
+    # This will simply initialize the requirer class and it won't perfom any action.
+
+    # Afterwards regardless of how you initialized the requirer you can call the
+    # provide_gateway_route_requirements method anywhere in your charm to update the requirer data.
+    # The method takes the same number of parameters as the requirer class.
+    # provide_gateway_route_requirements(name=, model=, ...)
```

Comment:
As a first step let's only support the first functionality ( directly specifying the parameters when instanciating the requirer class ) for now. 

We can add another class ( GatewayRouteDynamicRequirer ) to provide the values dynamically later when we implement the full relation